src/utils.rs

```diff
@@ -71,27 +74,83 @@ mod wasm {
     pub(crate) struct Mesh {
         pub(crate) kind: MeshKind,
         pub(crate) path: String,
-        pub(crate) data: String,
+        data: Data,
+    }
+
+    impl Mesh {
+        pub(crate) fn decode(data: &str) -> Result<Self> {
+            let mut mesh: Self = serde_json::from_str(data).map_err(|e| e.to_string())?;
+            match &mesh.data {
+                Data::None => {}
+                Data::Base64(s) => {
+                    mesh.data = Data::Bytes(base64::decode(s).map_err(|e| e.to_string())?);
+                }
+                Data::Bytes(_) => unreachable!(),
+            }
+            Ok(mesh)
+        }
+
+        pub(crate) fn reader(&self) -> Option<Cursor<&[u8]>> {
+            match &self.data {
+                Data::None => None,
+                Data::Bytes(bytes) => Some(Cursor::new(bytes)),
+                Data::Base64(_) => unreachable!(),
+            }
+        }
+
+        pub(crate) fn string(&self) -> Option<&str> {
+            match &self.data {
+                Data::None => None,
+                Data::Bytes(s) => str::from_utf8(s).ok(),
+                Data::Base64(..) => unreachable!(),
+            }
+        }
+    }
+
+    #[derive(Serialize, Deserialize)]
+    pub(crate) enum Data {
```

Comment:
Data is not a good name generally.
Could you add some information to this, like `MeshData` if this were only for meshes? (If you agree with this)

Reply:
renamed to MeshData